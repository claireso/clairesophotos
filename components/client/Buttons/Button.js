import React from 'react'
import PropTypes from 'prop-types'

import { StyledButton } from './Buttons.styles'

const Button = ({ children, ...props }) => {
  return <StyledButton {...props}>{children}</StyledButton>
}

Button.propTypes = {
  children: PropTypes.string.isRequired
}

export default Button
