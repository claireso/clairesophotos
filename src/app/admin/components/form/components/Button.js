import React from 'react'
import styled from 'styled-components'
import Loader from '@common/components/Loader'

export const SubmitButton = styled.input.attrs({
  type: 'submit'
})`
  appearence: none;
  background: var(--primary);
  border: none;
  color: white;
  cursor: pointer;
  display: block;
  font-size: 1.4rem;
  margin: calc(var(--gutter) * 2) 0 0;
  outline: none;
  padding: 1.5rem 5rem;
  width: 100%;
  transition: background 150ms ease-out;

  &:hover {
    background: var(--primary-lighter);
  }
`

export const SubmitButtonLoading = styled(SubmitButton).attrs({
  children: <Loader />
})`
  padding: 1.8rem 5rem 1.9rem;

  ${Loader} {
    margin: 0 auto;

    &:after {
      background: #fff;
    }
  }
`

export default props =>
  props.isLoading ? (
    <SubmitButtonLoading as="div" />
  ) : (
    <SubmitButton {...props} />
  )
