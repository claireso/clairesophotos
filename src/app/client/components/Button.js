import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const StyledDefaultButton = styled.button`
  appearance: none;
  background: transparent;
  border: none;
  cursor: pointer;
  outline: none;
  padding: 1rem 1.2rem 1.1rem;
  text-decoration: none;
`

const StyledButton = styled(StyledDefaultButton)`
  background: white;
  border: 1px solid var(--gray-1);
  display: inline-block;
  font-family: 'Roboto', Arial, sans-serif;
  width: 4.2rem;
  color: var(--text);
  font-size: 1.6rem;
  font-weight: 700;
  will-change: border-color;
  transition: border-color 150ms ease-out;

  &:hover {
    border-color: var(--gray-3);
  }
`

const StyledButtonIcon = styled(StyledDefaultButton)``

export const Button = ({ label, ...props }) => {
  return <StyledButton {...props}>{label}</StyledButton>
}

Button.propTypes = {
  label: PropTypes.string.isRequired
}

export const ButtonIcon = ({ children, ...props }) => {
  return <StyledButtonIcon {...props}>{children}</StyledButtonIcon>
}

ButtonIcon.propTypes = {
  children: PropTypes.element.isRequired
}
