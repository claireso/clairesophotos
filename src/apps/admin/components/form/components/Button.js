import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Loader from '@common/components/Loader'

export const SubmitButton = styled.input.attrs(() => ({
  type: 'submit'
}))`
  appearence: none;
  background: var(--primary);
  border: none;
  border-radius: 3.6rem;
  color: white;
  cursor: pointer;
  display: block;
  font-size: 1.4rem;
  margin: 3rem auto 0;
  outline: none;
  padding: 1.5rem 5.5rem;
  transition: background 150ms ease-out;

  &:hover {
    background: var(--primary-lighter);
  }
`

export const SubmitButtonLoading = styled.div.attrs(() => ({
  children: <Loader />
}))`
  margin: 3rem 0 0;
  padding: 1.8rem 0 1.9rem;

  ${Loader} {
    margin: 0 auto;
  }
`

const Button = props =>
  props.isLoading ? (
    <SubmitButtonLoading as="div" />
  ) : (
    <SubmitButton {...props} />
  )

Button.propTypes = {
  isLoading: PropTypes.bool
}

export default Button
