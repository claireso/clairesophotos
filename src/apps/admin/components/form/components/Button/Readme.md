### Examples

#### Basic usage

```js
<Button value="Send" />
```

#### Processing submit button

```js
<Button value="Send" isLoading />
```