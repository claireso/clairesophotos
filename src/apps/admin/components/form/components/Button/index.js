import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import Loader from '@common/components/Loader'

export const SubmitButton = styled.input.attrs(() => ({
  type: 'submit'
}))`
  appearance: none;
  background: var(--primary);
  border: none;
  border-radius: 3.6rem;
  color: white;
  cursor: pointer;
  display: block;
  font-size: 1.4rem;
  line-height: 1;
  margin: 2.8rem auto 0;
  outline: none;
  padding: 1.6rem 5.2rem;
  transition: background 150ms ease-out;

  &:hover {
    background: var(--primary-lighter);
  }
`

export const SubmitButtonLoading = styled.div.attrs(() => ({
  children: <Loader />
}))`
  margin: 2.8rem 0 0;
  padding: 1.8rem 0 1.8rem;

  ${Loader} {
    margin: 0 auto;
  }
`

/**
 * @visibleName SubmitButton
 */
const Button = props =>
  props.isLoading ? (
    <SubmitButtonLoading as="div" />
  ) : (
    <SubmitButton {...props} />
  )

Button.propTypes = {
  isLoading: PropTypes.bool
}

export default Button
