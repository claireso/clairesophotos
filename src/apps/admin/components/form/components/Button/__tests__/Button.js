import React from 'react'
import { render } from '@testing-library/react'

import Button from '../index'

describe('<Button />', () => {
  test('should render button', () => {
    const { container } = render(<Button value="Submit" />)

    expect(container).toMatchSnapshot()
  })

  test('should render loader', () => {
    const { container } = render(<Button value="Submit" isLoading />)

    expect(container).toMatchSnapshot()
  })
})
