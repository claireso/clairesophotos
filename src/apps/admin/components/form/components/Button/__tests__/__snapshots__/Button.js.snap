// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render button 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: var(--primary);
  border: none;
  border-radius: 3.6rem;
  color: white;
  cursor: pointer;
  display: block;
  font-size: 1.4rem;
  line-height: 1;
  margin: 2.8rem auto 0;
  outline: none;
  padding: 1.6rem 5.2rem;
  -webkit-transition: background 150ms ease-out;
  transition: background 150ms ease-out;
}

.c0:hover {
  background: var(--primary-lighter);
}

<div>
  <input
    class="c0"
    type="submit"
    value="Submit"
  />
</div>
`;

exports[`<Button /> should render loader 1`] = `
.c2 {
  margin: 2rem 0;
  -webkit-animation: jitEcr 800ms ease-in-out infinite alternate;
  animation: jitEcr 800ms ease-in-out infinite alternate;
}

.c2:after {
  background: var(--primary);
  border-radius: 50%;
  content: '';
  display: block;
  height: 1rem;
  width: 1rem;
  margin: 0 auto;
}

.c0 {
  margin: 2.8rem 0 0;
  padding: 1.8rem 0 1.8rem;
}

.c0 .c1 {
  margin: 0 auto;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 c2"
    />
  </div>
</div>
`;
